src/pip/_internal/cli/req_command.py

```diff
@@ -158,6 +160,33 @@ def handle_pip_version_check(self, options):
 ]
 
 
+def warn_if_run_as_root():
+    # type: () -> None
+    """Output a warning for sudo users on Unix.
+
+    In a virtual environment, sudo pip still writes to virtualenv.
+    On Windows, users may run pip as Administrator without issues.
+    This warning only applies to Unix root users outside of virtualenv.
+    """
+    if running_under_virtualenv():
+        return
+    if not hasattr(os, "getuid"):
+        return
+    # On Windows, there are no "system managed" Python packages. Installing as
+    # Administrator via pip is the correct way of updating system environments.
+    #
+    # We choose sys.platform over utils.compat.WINDOWS here to enable Mypy platform
+    # checks: https://mypy.readthedocs.io/en/stable/common_issues.html
+    if sys.platform == "win32" or sys.platform == "cygwin":
```

Comment:
Why are we doing this for cygwin? I'd have imagined that cygwin counts as a "system package manager" for the purposes of a cygwin build of Python. But having said that, I know very little about cygwin and I'm happy to assune it's OK unless cygwin users complain that they don't like it.

Reply:
It’s been a looong while since I last used Cygwin, but IIRC it does not have a notion of superusers, so it makes sense to skip the checks in it. But it’s probably a good idea to not over-anticipate things and wait for complaints instead, since this is just a line of message anyway.